ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/ScanConfigs/ViewScanConfigDetail.tsx

```diff
@@ -59,27 +92,57 @@ function ViewScanConfigDetail({
             <Divider component="div" />
             <PageSection variant="light" padding={{ default: 'noPadding' }}>
                 {!isLoading && !error && scanConfig && (
-                    <Flex
-                        alignItems={{ default: 'alignItemsCenter' }}
-                        className="pf-u-py-lg pf-u-px-lg"
-                    >
-                        <FlexItem flex={{ default: 'flex_1' }}>
-                            <Title headingLevel="h1">{scanConfig.scanName}</Title>
-                        </FlexItem>
-                        {hasWriteAccessForCompliance && (
-                            <FlexItem align={{ default: 'alignRight' }}>
-                                <Button
-                                    variant="primary"
-                                    component={LinkShim}
-                                    href={`${generatePath(complianceEnhancedScanConfigDetailPath, {
-                                        scanConfigId: scanConfig.id,
-                                    })}?action=edit`}
-                                >
-                                    Edit scan schedule
-                                </Button>
+                    <>
+                        <Flex
+                            alignItems={{ default: 'alignItemsCenter' }}
+                            className="pf-u-py-lg pf-u-px-lg"
+                        >
+                            <FlexItem flex={{ default: 'flex_1' }}>
+                                <Title headingLevel="h1">{scanConfig.scanName}</Title>
                             </FlexItem>
+                            {hasWriteAccessForCompliance && (
+                                <FlexItem align={{ default: 'alignRight' }}>
+                                    <Button
+                                        variant="secondary"
+                                        component={Button}
+                                        onClick={onTriggerRescan}
+                                        isLoading={isTriggeringRescan}
+                                        isDisabled={!scanConfig || isTriggeringRescan}
```

Comment:
What are pro and con of `!scanConfig` here when conditional rendering includes `scanConfig` above?